### ISSUE TITLE
Cluster Autoscaler fails to scale up AWS Managed Node Group which has multiple instance_types

### ISSUE THREAD

**Issue by pre** [kubernetes/autoscaler#3856]
**Which component are you using?**: cluster-autoscaler 

**What version of the component are you using?**: v1.20.0 (installed via helm-chart 9.4.0)

```
extraArgs:
  logtostderr: true
  stderrthreshold: info
  v: 4
  write-status-configmap: true
  skip-nodes-with-local-storage: false
  scale-down-enabled: true
  balance-similar-node-groups: true
  min-replica-count: 2
  scale-down-utilization-threshold: 0.5
  scale-down-non-empty-candidates-count: 5
  max-node-provision-time: 15m0s
  scan-interval: 10s
  scale-down-delay-after-add: 1m
  scale-down-delay-after-delete: 0s
  scale-down-delay-after-failure: 3m
  scale-down-unneeded-time: 1m
```

**What k8s version are you using (`kubectl version`)?**: EKS 1.18

**What environment is this in?**: EKS 1.18

**What did you expect to happen?**:

We have multiple Managed Node Groups which have different instance types in each of them. 

For example, there is one compute oriented Managed Node Group which provides all the variants of EC2 c5 instance types:
```
  instance_types = ["c5.large", "c5.xlarge", "c5.2xlarge", "c5.4xlarge", "c5.9xlarge", "c5.12xlarge", "c5.18xlarge", "c5.24xlarge"]
```

[It seems](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#how-does-scale-up-work) that cluster-autoscaler does not understand that a Managed Node Group can provide with different instance types. 

Let's assume that cluster is already running one type of an instance from this Managed Node Group, for example `c5.large`. 
When a Deployment requests an instance type from that same Managed Node Group, but type which _is not_ yet available in the cluster, the pods stay in Pending stage indefinitely. (nodeSelector for eg. `c5.24xlarge`).

[It seems](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#how-does-scale-up-work) that cluster-autoscaler does not understand that the Managed Node Group, which has already provided `c5.large`, can _also_ provide eg. `c5.24xlarge`. 

Hence, the Pods which requested `c5.24xlarge` do not trigger scale up of a new machine, because cluster-autoscaler thinks that this Managed Node Group only provides `c5.large` instances, although the Managed Node Group can provide _any_ `c5.*`instance types (see array above). 

The workaround for this could be to create a separate Managed Node Group for each of the instance types, each group containing only a single instance_type. However, due to a [limitation in the Managed Node Group](https://github.com/aws/containers-roadmap/issues/724), you cannot currently have `min_size: 0`. This means that each Managed Node Group would need to always be running at least one node. 

TL;DR It seems that cluster-autoscaler is not currently compatible with EKS Managed Node Groups with multiple instance_types. Please correct me if I'm wrong! I'd like to find out a workaround for this issue.

**Comment by pre** [comment]
To be precise, the cluster-autoscaler does scale up different instance_types from a Managed Node Group. However, the instance type cannot be directed with `nodeSelector`. 

For example: Scaling up 50 pods which eg. `requests.cpu: 1` will scale up eg. `c5.12xlarge` instance(s). 

What fails is that when an explicit node is requested, cluster-autoscaler fails to bring it up. 

This is most easily replicated by requesting the largest node type `c5.24xlarge` because it practically never exists in the cluster before hand (and is not scaled up by cluster-autoscaler even there is the nodeSelector for it).

**Comment by TBBle** [comment]
From the [FAQ](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#what-are-the-key-best-practices-for-running-cluster-autoscaler):
> What are the key best practices for running Cluster Autoscaler?
> * [...] All nodes within the same node group should have the same capacity, labels and system pods running on them.

Also from the [AWS docs](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/aws/README.md#Using-Mixed-Instances-Policies-and-Spot-Instances)
> Using Mixed Instances Policies and Spot Instances
> [...] Note that the instance types should have the same amount of RAM and number of CPU cores, since this is fundamental to CA's scaling calculations. Using mismatched instances types can produce unintended results. See an example below.
> ...
> * For the same workload or for the generic capacity in your cluster, you can also create more node groups with a vCPU/Mem ratio that is a good fit for your workloads, but from different instance sizes. For example: Node group 1: m5.xlarge, m5a.xlarge, m5d.xlarge, m5ad.xlarge, m4.xlarge. Node group 2: m5.2xlarge, m5a.2xlarge, m5d.2xlarge, m5ad.2xlarge, m4.2xlarge. This approach increases the chance of achieving your desired scale at the lowest cost by tapping into many Spot capacity pools.

So this seems to be a pretty core expectation of Cluster Autoscaler, that the node groups are homogenous, and I can't see any way to change that without reworking Cluster Autoscaler's core behaviour, and introducing some kind of extra decision-making mechanism _within_ a node group, which CA does not currently deal with in any way.

**Comment by pre** [comment]
I understand that AWS Managed Node Groups with heterogenous instance types is a fairly new feature and cluster-autoscaler was designed for homogenous node groups. 

However, even the [official AWS documentation](https://docs.aws.amazon.com/eks/latest/userguide/managed-node-groups.html) recommends using multiple instance types per Managed Node Group. 

As of today, this means that cluster-autoscaler is not compatible with AWS Managed Node Groups. 

I propose that this would be documented somewhere, because as of today, you'll run into a surprise when cluster-autoscaler does not even attempt to scale up new nodes even though it sees that there are Pods in Pending state because the requested nodes are not available.

**Comment by Legion2** [comment]
@pre it is documented in the section "Considerations for selecting a capacity type"

> Within a managed node group, if you're using the Cluster Autoscaler, we recommend using a flexible set of instance types with the same amount of vCPU and memory resources to ensure that the nodes in your cluster scale as expected
